### PR TITLE
Carry 24492: Remove swarm inspect and use info instead

### LINKED ISSUE #24148: `docker swarm inspect` and `docker info` inconsistencies
Both `docker info` and `docker swarm inspect` display information regarding the swarm.

They do so in a different way.

There are also other inconsistencies in `swarm inspect` (#23689, #24122).

/cc @stevvooe @aanand @dnephin @tonistiigi

### PR DESCRIPTION
**Pull request by tiborvass**
Carrying #24492

Vendored https://github.com/docker/engine-api/pull/334

Closes #24492
Closes #24148

### CODE REVIEW

**Comment by mlaventure** [comment]
LGTM,

but docker info output is becoming quite large, for readability we may want to consider introducing subcommand to info later on (e,g `docker info swarm`) to limit the scope of the displayed information. The original `docker info` would still display everything (easier for issue reporting)

File: api/client/system/info.go [hunk 1]
```diff
@@ -83,8 +85,20 @@ func runInfo(dockerCli *client.DockerCli) error {
 		}
 		fmt.Fprintf(dockerCli.Out(), " Is Manager: %v\n", info.Swarm.ControlAvailable)
 		if info.Swarm.ControlAvailable {
+			fmt.Fprintf(dockerCli.Out(), " ClusterID: %s\n", info.Swarm.Cluster.ID)
 			fmt.Fprintf(dockerCli.Out(), " Managers: %d\n", info.Swarm.Managers)
 			fmt.Fprintf(dockerCli.Out(), " Nodes: %d\n", info.Swarm.Nodes)
+			fmt.Fprintf(dockerCli.Out(), " Name: %s\n", info.Swarm.Cluster.Spec.Annotations.Name)
+			fmt.Fprintf(dockerCli.Out(), " Orchestration:\n")
+			fmt.Fprintf(dockerCli.Out(), "  Task History Retention: %d\n", info.Swarm.Cluster.Spec.Orchestration.TaskHistoryRetentionLimit)
```
Comment (aaronlehmann) [inline api/client/system/info.go]: either "Task History Retention Limit: %d" or "Task History Retention: %d tasks"

File: api/client/system/info.go [hunk 2]
```diff
@@ -83,8 +85,20 @@ func runInfo(dockerCli *client.DockerCli) error {
 		}
 		fmt.Fprintf(dockerCli.Out(), " Is Manager: %v\n", info.Swarm.ControlAvailable)
 		if info.Swarm.ControlAvailable {
+			fmt.Fprintf(dockerCli.Out(), " ClusterID: %s\n", info.Swarm.Cluster.ID)
 			fmt.Fprintf(dockerCli.Out(), " Managers: %d\n", info.Swarm.Managers)
 			fmt.Fprintf(dockerCli.Out(), " Nodes: %d\n", info.Swarm.Nodes)
+			fmt.Fprintf(dockerCli.Out(), " Name: %s\n", info.Swarm.Cluster.Spec.Annotations.Name)
```
Comment (aaronlehmann) [inline api/client/system/info.go]: Remove this line. This is always set to "default" and will be confusing since it's not clear it's the cluster name.

**Comment by aaronlehmann** [comment]
Some minor suggestions, otherwise LGTM.

**Comment by stevvooe** [comment]
LGTM

**Comment by vieux** [comment]
LGTM, I agree with @aaronlehmann's suggestions

**Comment by tiborvass** [comment]
@aaronlehmann fixed

**Comment by aaronlehmann** [comment]
Merge away when CI is ready.

**Comment by tiborvass** [comment]
Win2lin has issues, it's unrelated

**Pull request merged**